app/src/main/java/com/hunguet123/searchImage/data/repository/PhotoRepositoryImpl.kt

```diff
@@ -0,0 +1,37 @@
+package com.hunguet123.searchImage.data.repository
+
+import com.hunguet123.searchImage.data.PhotoRepository
+import com.hunguet123.searchImage.data.model.Photo
+import com.hunguet123.searchImage.data.source.PhotoDataSource
+import com.hunguet123.searchImage.data.source.local.room.PhotoRoomDatabase
+
+class PhotoRepositoryImpl(
+    private val remote : PhotoDataSource.Remote,
+    private val photoRoomDatabase: PhotoRoomDatabase
+) : PhotoRepository {
+    override suspend fun getImages(q: String?): List<Photo> {
+        val listPhoto = remote.getImages(q).hits ?: emptyList()
+        val listPhotoLocal = photoRoomDatabase.photoDao().getAllPhotos()
+        for (photo in listPhoto) {
+            for (photoLocal in listPhotoLocal) {
+                if (photo.id == photoLocal.id) {
+                    photo.favorite = photoLocal.favorite
+                }
+            }
+        }
```

Comment:
1. giảm độ phức tạp xuống O(n) thôi, tạo 1 set of photo id 
2. repository quản lý việc lấy dữ liệu từ remote, local thôi. Việc mix dữ liệu giữa 2 nguồn em đẩy lên ViewModel xử lý. logic 
3. thường thi logic người ta sẽ viết ở ViewModel để viết unit test, những tầng data người ta thường check mock data